TuneBot/Modules/TrackModule.cs

```diff
@@ -2,331 +2,282 @@
         }
 
-        [Command("np")]
-        [Summary("Gets current/last scrobbled track from Last.fm.")]
+        [SlashCommand("np", "Gets current/last scrobbled track from Last.fm.")]
         public async Task NowPlaying()
         {
-            using var typing = Context.Channel.EnterTypingState();
             var userId = Context.User.Id;
 
-            string? lastFmUserName = null;
+            string? lastFmUserName = await _dbService.GetLastFmUsername(userId);
+
+            if (string.IsNullOrEmpty(lastFmUserName))
+            {
+                await RespondAsync(
+                    "No Last.fm user name found. Use !npset <username> to set your Last.fm user name");
+                return;
+            }
 
-            using (var connection = new SqliteConnection("Data Source=data.db"))
+            using var client = new LastfmClient(_configuration["LastFmKey"], _configuration["LastFmSecret"]);
             {
-                await connection.OpenAsync();
-                using (var command = connection.CreateCommand())
+                try
                 {
-                    command.CommandText = "SELECT * FROM users WHERE id=$id";
-                    command.Parameters.AddWithValue("$id", userId);
+                    var track = await client.User.GetRecentScrobbles(lastFmUserName, count: 1);
 
-                    using (var reader = command.ExecuteReader())
-                    {
-                        while (reader.Read())
-                        {
-                            var isNull = await reader.IsDBNullAsync(1);
+                    var lastTrack = track.FirstOrDefault();
 
-                            lastFmUserName = isNull == true ? null : reader.GetString(1);
-                        }
+                    if (lastTrack is null)
+                    {
+                        await RespondAsync("No now playing data found for Last.fm user");
+                        return;
                     }
-                }
-            }
 
-            if (lastFmUserName != null)
-            {
-                using var client = new LastfmClient(config["LastFmKey"], config["LastFmSecret"]);
+                    var songName = lastTrack.ArtistName + " " + lastTrack.Name;
 
-                var track = await client.User.GetRecentScrobbles(lastFmUserName, count: 1);
+                    var spotifyTrack = await SearchSpotifyAsync(songName);
 
-                var lastTrack = track.FirstOrDefault();
+                    var embed = await CreateEmbedAsync(spotifyTrack!);
 
-                if (lastTrack != null)
-                {
-                    var songName = lastTrack.ArtistName + " " + lastTrack.Name;
+                    if (embed is null)
+                    {
+                        await RespondAsync("Data for now playing not found");
+                        return;
+                    }
+
+                    var youtubeLink = await SearchYouTubeAndGetLinkAsync(songName);
```

Comment:
We're awaiting `SearchSpotifyAsync` and then later on awaiting `SearchYouTubeAndGetLinkAsync` which means that these request run in series and can cause the command to take longer.. We can kick off the YouTube methid before awaiting the Spotify method, then when we need the YouTube result, await it and use the result.